contentcuration/contentcuration/frontend/channelList/views/ChannelSet/ChannelSetModal.vue

```diff
@@ -350,44 +383,54 @@
       },
       confirmCancel() {
         if (this.isNew) {
-          return this.deleteChannelSet(this.channelSet).then(this.close);
+          if (this.channelSet && this.channelSet.id) {
+            return this.deleteChannelSet(this.channelSet).then(this.close);
+          } else {
+            this.close();
+          }
+        } else {
+          this.close();
         }
-        this.close();
       },
+
       close() {
         this.changed = false;
         this.showUnsavedDialog = false;
         this.diffTracker = {};
         this.$router.push({ name: RouteNames.CHANNEL_SETS });
       },
+
       verifyChannelSet(channelSetId) {
-        return new Promise((resolve, reject) => {
-          // Check if we already have the channel locally
-          if (this.getChannelSet(channelSetId)) {
-            this.setup();
-            resolve();
-            return;
-          }
+        if (!channelSetId || channelSetId === 'new') {
```

Comment:
I can't seem to see where channelSetId is set to `new` for the `or` part of this condition to be successful?